aptos-move/framework/aptos-framework/sources/HashMap.move

```diff
@@ -0,0 +1,281 @@
+module AptosFramework::HashMap {
+    use Std::Errors;
+    use Std::Vector;
+    use Std::Hash::sip_hash;
+    use AptosFramework::Table::{Self, Table};
+
+    const MAX_LOAD_PER_BUCKET: u64 = 10;
+    const SPLIT_THRESHOLD: u64 = 75;
+
+    /// Not found in the table;
+    const ENOT_FOUND: u64 = 0;
+    /// Initial capacity needs to be power of two.
+    const ENOT_POWER_OF_TWO: u64 = 1;
+    /// Destroy non-empty hashmap.
+    const ENOT_EMPTY: u64 = 2;
+
+    /// HashMap entry contains both the key and value.
+    struct Entry<K, V> has store {
+        hash: u64,
+        key: K,
+        value: V,
+    }
+
+    /// A hashmap implementation based on linear hashing. (https://www.cs.cmu.edu/afs/cs.cmu.edu/user/christos/www/courses/826-resources/PAPERS+BOOK/linear-hashing.PDF)
```

Comment:
Need more documentation on implementation details of the table, bucket management etc,

Reply:
I don't think I can describe it well in a short page, the paper describes it. maybe I can add the wikipedia too